spec/lib/html_cleaner_spec.rb

```diff
@@ -1104,4 +1104,90 @@
       expect(add_break_between_paragraphs(original)).to eq(result)
     end
   end
+
+  describe "strip_images" do
+    let(:result) { "Hi!  Bye" }
+
+    context "without keep_src" do
+      it "removes the img tag entirely when the src uses double quotes" do
+        string = 'Hi! <img src="http://example.org/image.png" /> Bye'
+        expect(strip_images(string)).to eq(result)
+      end
+
+      it "removes the img tag entirely when the src uses single quotes" do
+        string = "Hi! <img src='http://example.org/image.png'> Bye"
+        expect(strip_images(string)).to eq(result)
+      end
+
+      it "removes the img tag entirely when the src uses mismatched quotes" do
+        string = "Hi! <img src=\"http://example.org/image.png'> Bye"
+        expect(strip_images(string)).to eq(result)
+      end
+
+      it "removes the img tag entirely when the src is missing" do
+        string = 'Hi! <img alt="a11y"> Bye'
+        expect(strip_images(string)).to eq(result)
+      end
+
+      it "removes the img tag entirely when the src is missing a closing quotation mark" do
+        string = 'Hi! <img src="http://example.org/image.png /> Bye'
+        expect(strip_images(string)).to eq(result)
+      end
+    end
+
+    context "with keep_src: false" do
+      it "removes the img tag entirely when the src uses double quotes" do
+        string = 'Hi! <img src="http://example.org/image.png" /> Bye'
+        expect(strip_images(string, keep_src: false)).to eq(result)
+      end
+
+      it "removes the img tag entirely when the src uses single quotes" do
+        string = "Hi! <img src='http://example.org/image.png'> Bye"
+        expect(strip_images(string, keep_src: false)).to eq(result)
+      end
+
+      it "removes the img tag entirely when the src uses mismatched quotes" do
+        string = "Hi! <img src=\"http://example.org/image.png'> Bye"
+        expect(strip_images(string, keep_src: false)).to eq(result)
+      end
+
+      it "removes the img tag entirely when the src is missing" do
+        string = 'Hi! <img alt="a11y"> Bye'
+        expect(strip_images(string, keep_src: false)).to eq(result)
+      end
+
+      it "removes the img tag entirely when the src is missing a closing quotation mark" do
+        string = 'Hi! <img src="http://example.org/image.png /> Bye'
+        expect(strip_images(string, keep_src: false)).to eq(result)
+      end
+    end
+
+    context "with keep_src: true" do
+      it "keeps the img src URL when the src uses double quotes" do
+        string = 'Hi! <img src="http://example.org/image.png" /> Bye'
+        result = "Hi! http://example.org/image.png Bye"
+        expect(strip_images(string, keep_src: true)).to eq(result)
+      end
+
+      it "removes the img tag entirely when the src uses single quotes" do
+        string = "Hi! <img src='http://example.org/image.png'> Bye"
```

Comment:
Yeah, idk. I originally treated them the same for consistency, even though I don't really like writing code just to make tests pass, but after Bilka's comment, I got more skittish about that approach. 

I think maybe we should have different behavior so it forces us to think about what's going on, e.g., if we're seeing the wrong behavior on production, that's a big yikes because it means the text isn't getting sanitized properly, and if we're seeing the wrong behavior in tests, is it because we're making test data in a way that's bypassing the sanitizer, or because there's actually a problem.

That said, I'm happy to take the path of least resistance and do `['"]` in the regex instead.

Reply:
I can think of ways to break this new regexp, from more spaces to newlines. But I don't think rewriting this regexp until I run out of obvious ways to break it will get us anywhere actually safe, that's why I suggested using a lib for it. The regexp will never quite protect from the case of some place accidentally doing `sanitize_field(strip_images())` (perhaps hidden behind or separated by steps of abstraction) instead of using the arg or the correct order. Or one of the myriad of other ways unsanitized content could end up in strip_images (like in tests) while still appearing safe at first glance.

But I see that the dynamic config option means it can't happen during the sanitize step and tossing the whole thing into a new nokogiri document for a second time in strip_images isn't a great solution either. Furthermore, this PR is not the right place to rework comments to store unsanitized content and only sanitize (and strip images during sanitization) for display.

So, I'd say to make a regexp that's readable/feels good to you and only secondary that it handles unexpected input. And then in the integration tests where things _should_ be sanitized beforehand, use an img tag that specifically would break the regexp if things run in the wrong order. That should be slightly more protection against the "big yikes" than the current test setup. Not good, but better 😭